### PR TITLE
feat(pie-input): DSW-1582 dispatch change event

### PR DESCRIPTION
**Pull request by jamieomaguire**
---
"@justeattakeaway/pie-switch": minor
---

[Changed] - Use new event wrapper function to standardise the change event structure

---
"pie-monorepo": minor
---

[Added] - Readme section for how we use events in our web components

---
"@justeattakeaway/pie-webc-core": minor
---

[Added] - new wrapNativeEvent function to standardise how we structure custom versions of native browser events such as 'change'

### CODE REVIEW

File: packages/components/README.md [hunk 1]
```diff
@@ -2,6 +2,22 @@
 
 This readme contains some general information that is relevant for all components. For specific information about a component, please refer to the readme in the component's directory.
 
+## Events
+Some components dispatch events that consuming applications can listen for. In some cases, we simply allow native events, such as `input` to bubble up from the component. However, in some cases we need to dispatch custom events. This could be due to the fact that the component does not have a native counter part, and so the event is unique to the component. Or it could be that the native event, such as `change`
```
Comment (fernandofranca) [inline packages/components/README.md:6]: Should it be spelled `counterpart`?

File: packages/components/pie-switch/src/index.ts [hunk 2]
```diff
@@ -94,13 +94,7 @@ export class PieSwitch extends FormControlMixin(RtlMixin(LitElement)) implements
     onChange (event: Event) {
         const { checked } = event?.currentTarget as HTMLInputElement;
         this.checked = checked;
-        const changedEvent = new CustomEvent(
-            ON_SWITCH_CHANGED_EVENT,
```
Comment (fernandofranca) [inline packages/components/pie-switch/src/index.ts:98]: There's a ESLint warning after this was removed: 'ON_SWITCH_CHANGED_EVENT' is defined but never used

File: packages/components/README.md (hunk 1 above)
Comment (jamieomaguire) [reply packages/components/README.md:6]: yes 😂 good spot!

**Review by fernandofranca — APPROVED** [review]

**Review by xander-marjoram — APPROVED** [review]

File: packages/components/pie-webc-core/src/functions/wrapNativeEvent.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+/**
+ * Wraps the native event in a custom event. This is useful for native events that have `composed` set to `false`
+ * as these cannot propagate outside of the shadow DOM.
```
Comment (ashleynolan) [inline packages/components/pie-webc-core/src/functions/wrapNativeEvent.ts:3]: Can be done in follow up, but can you add the link to the docs/article you showed on this (or to our wiki/readme section if it's in there)

**Pull request merged**